src/instruments/instruments.controller.ts

```diff
@@ -57,14 +59,20 @@ export class InstrumentsController {
         await this.instrumentsService.create(createInstrumentDto);
       return instrument;
     } catch (error) {
-      let message;
-      if (error instanceof Error) message = error.message;
-      else message = String(error);
-      // we'll proceed, but let's report it
-      throw new HttpException(
-        `Instrument with the same unique name already exists: ${message}`,
-        HttpStatus.BAD_REQUEST,
-      );
+      if ((error as MongoError).code === 11000) {
+        throw new ConflictException(
+          "Instrument with the same unique name already exists",
+        );
+      } else {
+        logger.logException(
+          "Something went wrong while creating instrument",
+          error,
+        );
+
+        throw new InternalServerErrorException(
+          "Something went wrong. Please try again later.",
+        );
```

Comment:
We don't need to seperate two logs here, all exceptions will be sent to the graylogs once enabled
Can we keep the original error object in InternalServerErrorException for debugging?